Projects/Checkers/Program.cs

```diff
@@ -0,0 +1,227 @@
+	Console.WriteLine("  across the board diagonally. Should a piece reach the other side of the board");
+	Console.WriteLine("  the piece becomes a king and can then move diagonally backwards as well as");
+	Console.WriteLine("  forwards.");
+	Console.WriteLine();
+	Console.WriteLine("  Pieces are captured by jumping over them diagonally. More than one enemy piece");
+	Console.WriteLine("  can be captured in the same turn by the same piece. If you can capture a piece");
+	Console.WriteLine("  you must capture a piece.");
+	Console.WriteLine();
+	Console.WriteLine("  Moves are selected with the arrow keys. Use the [enter] button to select the");
+	Console.WriteLine("  from and to squares. Invalid moves are ignored.");
+	Console.WriteLine();
+	Console.WriteLine("  Press a number key to choose number of human players:");
+	Console.WriteLine("    [0] Black (computer) vs White (computer)");
+	Console.WriteLine("    [1] Black (human) vs White (computer)");
+	Console.Write("    [2] Black (human) vs White (human)");
+
+	int? humanPlayerCount = null;
+	while (humanPlayerCount is null)
+	{
+		Console.CursorVisible = false;
+		switch (Console.ReadKey(true).Key)
+		{
+			case ConsoleKey.D0 or ConsoleKey.NumPad0: humanPlayerCount = 0; break;
+			case ConsoleKey.D1 or ConsoleKey.NumPad1: humanPlayerCount = 1; break;
+			case ConsoleKey.D2 or ConsoleKey.NumPad2: humanPlayerCount = 2; break;
+		}
+	}
+	return new Game(humanPlayerCount.Value);
+}
+
+void RunGameLoop(Game game)
+{
+	while (game.Winner is null)
+	{
+		Player currentPlayer = game.Players.First(player => player.Color == game.Turn);
+		if (currentPlayer is not null && currentPlayer.IsHuman)
+		{
+			while (game.Turn == currentPlayer.Color)
+			{
+				(int X, int Y)? selectionStart = null;
+				(int X, int Y)? from = game.Board.Aggressor is not null ? (game.Board.Aggressor.X, game.Board.Aggressor.Y) : null;
+				List<Move> moves = game.Board.GetPossibleMoves(game.Turn);
+				if (moves.Select(move => move.PieceToMove).Distinct().Count() is 1)
+				{
+					Move must = moves.First();
+					from = (must.PieceToMove.X, must.PieceToMove.Y);
+					selectionStart = must.To;
+				}
+				(int X, int Y)? to = null;
+				while (from is null)
+				{
+					from = HumanMoveSelection(game);
+					selectionStart = from;
+				}
+				to = HumanMoveSelection(game, selectionStart: selectionStart, from: from);
+				Piece? piece = null;
+				if (from is not null)
+				{
+					piece = game.Board[from.Value.X, from.Value.Y];
+				}
+				if (piece is null || piece.Color != game.Turn)
+				{
+					from = null;
+					to = null;
+				}
+				if (from is not null && to is not null)
+				{
+					Move? move = game.Board.ValidateMove(game.Turn, from.Value, to.Value);
+					if (move is not null &&
+						(game.Board.Aggressor is null || move.PieceToMove == game.Board.Aggressor))
+					{
+						game.PerformMove(move);
+					}
+				}
+			}
+		}
+		else
+		{
+			List<Move>? moves = game.Board.GetPossibleMoves(game.Turn);
```

Comment:
A style thing (here and elsewhere) but I prefer var as it reduces the noise and you rarely care about the type when reading the code (which happens far more often than writing).

Reply:
I may be in the minority, but I personally hate `var` with a passion. I think it actually hurts readability. Some people may only ever read the code on GitHub, and if `var` is used then they can't just `Go To Definition` (since that isn't supported on GitHub).

However, if you feel strongly about it, feel free to change it before the pull request is merge or open a new pull request in the future.